### PR TITLE
opengl: triangle

### PR DESCRIPTION
**Pull request by pdoane**
- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: src/shader/CodeGen.zig [hunk 1]
```diff
@@ -20,10 +22,12 @@ pub fn generate(
     air: *const Air,
     out_lang: Language,
     debug_info: DebugInfo,
+    entrypoint: ?[*:0]const u8,
```
Comment (alichraghi) [inline src/shader/CodeGen.zig:25]: Why is this needed here? We already have custom entry point field here
https://github.com/hexops/mach-dusk/blob/709a94af18695d8c68d05137f59f11c7703c840e/src/shader/Air.zig#L34

Comment (pdoane) [reply src/shader/CodeGen.zig:25]: This will likely all change when considering offline compilation, but I'm assuming we want semantic analysis to be complete after createShaderModule.  The conversion to target language needs to happen late - I think all 3 high-level languages will eventually need it to be done during pipeline creation without WGSL changes.

Or maybe I'm not following how you expect it to be used?

Comment (alichraghi) [reply src/shader/CodeGen.zig:25]: No i was just trying to understand why it's needed.

**Review by slimsag — APPROVED** [review]
